net: lib: wifi_mgmt_ext (#9261)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

Wi-Fi
-----

* Added:

  * wifi_credentials library to store credentials

>   * wifi_mgmt_ext library to provide an autoconnect command based on wifi_credentials

Reviewer: This sounds like we have some new libraries, but where is the actual documentation for them?

Wi-Fi
-----

* Added:

>   * wifi_credentials library to store credentials
>   * wifi_mgmt_ext library to provide an autoconnect command based on wifi_credentials

Reviewer: ```suggestion
  * :ref:`lib_wifi_credentials` library to store credentials.
  * :ref:`wifi_mgmt_ext` library to provide an autoconnect command based on Wi-Fi credentials.
```

  * New sample :ref:`wifi_sr_coex_sample` demonstrating Wi-Fi Bluetooth LE coexistence.
  * :ref:`ug_wifi` document.

>   * Sample demonstrating Wi-Fi Bluetooth LE coexistence

Reviewer: We already have this above.

Author: removed this

  * :ref:`wifi_sr_coex_sample` sample demonstrating Wi-Fi Bluetooth LE coexistence.

* Changed:

>   * The :ref:`wifi_shell` sample now makes use of the :ref:`lib_wifi_credentials` and :ref:`wifi_mgmt_ext` libraries.

Reviewer: ```suggestion
  * The :ref:`wifi_shell_sample` sample now makes use of the :ref:`lib_wifi_credentials` and :ref:`wifi_mgmt_ext` libraries.
```
This clears the doc build failure. Dunno what to do about the conformance failure.

  * :ref:`wifi_station_sample` sample that demonstrates how to connect the Wi-Fi station to a specified access point.
  * :ref:`wifi_provisioning` sample that demonstrates how to provision a device with Nordic Semiconductor's Wi-Fi chipsets over Bluetooth® Low Energy.

* Changed:

>   * :ref:`wifi_shell` sample now makes use of wifi_credentials and wifi_mgmt_ext.

Reviewer: If we have such new libraries, we need to have docs for them and link to them.

Author: @peknis I added some docs now. Could you have a look?

  * :ref:`wifi_sr_coex_sample` sample demonstrating Wi-Fi Bluetooth LE coexistence.

* Changed:

>   * The :ref:`wifi_shell_sample` sample now makes use of the :ref:`lib_wifi_credentials` and :ref:`wifi_mgmt_ext` libraries.

Reviewer: ```suggestion
  * The :ref:`wifi_shell_sample` sample now uses the :ref:`lib_wifi_credentials` and :ref:`wifi_mgmt_ext` libraries.
```

Reviewer: Looks like you did not address this one.

Author: sorry, I meant to include it

  * :ref:`wifi_station_sample` sample that demonstrates how to connect the Wi-Fi station to a specified access point.
  * :ref:`wifi_provisioning` sample that demonstrates how to provision a device with Nordic Semiconductor's Wi-Fi chipsets over Bluetooth® Low Energy.

* Changed:

>   * :ref:`wifi_shell` sample now makes use of wifi_credentials and wifi_mgmt_ext.

Author: I added something like that now.
